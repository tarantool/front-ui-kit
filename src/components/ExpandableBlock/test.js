import React from 'react';
import renderer from 'react-test-renderer';
import { ExpandableBlock } from './index';

// eslint-disable-next-line max-len
const content = 'Pipeline execution failed: Function "bad_task" in pipeline "bad_task":\nfunction_call_error: [string "bad_task"]:1: failure\nstack traceback:\n\t[string "bad_task"]:1: in main chunk\n\t[C]: in function \'xpcall\'\n\t...\/errors.lua:148: in function \'pcall\'\n\t...\/init.lua:109: in function \'call\'\n\t...\/pipeline\/init.lua:21: in function \'call_function\'\n\t...pipeline\/init.lua:63: in function \'run\'\n\t...\/runner\/server.lua:70: in function <...\/runner\/server.lua:68>\nstack traceback:\n\t...\/pipeline\/init.lua:25: in function \'call_function\'\n\t...\/pipeline\/init.lua:63: in function \'run\'\n\t...\/runner\/server.lua:70: in function <...s\/runner\/server.lua:68>';


describe('Table', () => {
  it('renders without more btm', () => {
    const tree = renderer.create(
      <>
        <ExpandableBlock
          content={content}
          visibleLines={100}
        />
      </>
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('renders with more btm', () => {

    const tree = renderer.create(
      <ExpandableBlock
        content={content}
        visibleLines={4}
      />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('renders with copy btn', () => {

    const tree = renderer.create(
      <ExpandableBlock
        content={content}
        visibleLines={4}
        textToCopy={content}
      />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
