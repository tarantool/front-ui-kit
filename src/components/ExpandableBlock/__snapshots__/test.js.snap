// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Table renders with copy btn 1`] = `
.emotion-0 {
  position: relative;
}

.emotion-1 {
  white-space: nowrap;
  border: none;
  border-radius: 4px;
  box-sizing: border-box;
  font-family: Inter,Arial,sans-serif;
  font-size: 14px;
  line-height: 22px;
  transition-timing-function: ease-in-out;
  transition-duration: 0.07s;
  transition-property: background-color,color;
  outline: none;
  cursor: pointer;
  -webkit-font-smoothing: antialiased;
  padding: 1px 4px;
  background-color: transparent;
  color: #5c606c;
  cursor: pointer;
  position: absolute;
  top: 0;
  right: 0px;
}

.emotion-1:hover,
.emotion-1:focus,
.emotion-1:active {
  transition-property: background-color,color,box-shadow;
}

.emotion-1:disabled,
.emotion-1:disabled:active,
.emotion-1:disabled:hover {
  cursor: default;
}

.emotion-1:focus,
.emotion-1:hover {
  background-color: #e6e7e8;
}

.emotion-1:disabled,
.emotion-1:disabled:active,
.emotion-1:disabled:hover {
  background-color: #fafafa;
  color: #d9dadd;
}

.emotion-1 svg {
  fill: #5c606c;
}

.emotion-1:disabled svg {
  fill: #d9dadd;
}

.emotion-2 {
  width: 16px;
  height: 16px;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  vertical-align: middle;
  width: 14px;
  height: 14px;
  width: 16px;
  height: 16px;
  fill: #5c606c;
  width: 16px;
  height: 16px;
  margin-bottom: 2px;
}

.emotion-3 {
  margin: 0;
  font-family: Inter,Arial,sans-serif;
  font-size: 14px;
  line-height: 22px;
  color: #000000;
  -webkit-font-smoothing: antialiased;
  position: relative;
  cursor: pointer;
  white-space: pre-wrap;
  margin-right: 30px;
}

.emotion-3 b {
  font-weight: 600;
}

.emotion-3::after {
  content: '';
  position: absolute;
  bottom: 0;
  left: 0;
  width: 100%;
  height: 30px;
  background: linear-gradient(180deg, rgba(255, 255, 255, 0) 0%, #FFFFFF 100%);
  pointer-events: none;
}

<div
  className="emotion-0"
>
  <button
    className="emotion-1"
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    onScroll={[Function]}
    placement="top"
    type="button"
  >
    <svg
      className="emotion-2"
      viewBox="0 0 16 16"
    >
      <use
        xlinkHref="#icon-copy"
      />
    </svg>
  </button>
  <div
    className="emotion-3"
    onClick={[Function]}
  >
    Pipeline execution failed: Function "bad_task" in pipeline "bad_task":
function_call_error: [string "bad_task"]:1: failure
stack traceback:
	[string "bad_task"]:1: in main chunk
  </div>
</div>
`;

exports[`Table renders with more btm 1`] = `
.emotion-0 {
  margin: 0;
  font-family: Inter,Arial,sans-serif;
  font-size: 14px;
  line-height: 22px;
  color: #000000;
  -webkit-font-smoothing: antialiased;
  position: relative;
  cursor: pointer;
  white-space: pre-wrap;
}

.emotion-0 b {
  font-weight: 600;
}

.emotion-0::after {
  content: '';
  position: absolute;
  bottom: 0;
  left: 0;
  width: 100%;
  height: 30px;
  background: linear-gradient(180deg, rgba(255, 255, 255, 0) 0%, #FFFFFF 100%);
  pointer-events: none;
}

<div
  className="emotion-0"
  onClick={[Function]}
>
  Pipeline execution failed: Function "bad_task" in pipeline "bad_task":
function_call_error: [string "bad_task"]:1: failure
stack traceback:
	[string "bad_task"]:1: in main chunk
</div>
`;

exports[`Table renders without more btm 1`] = `
.emotion-0 {
  margin: 0;
  font-family: Inter,Arial,sans-serif;
  font-size: 14px;
  line-height: 22px;
  color: #000000;
  -webkit-font-smoothing: antialiased;
  white-space: pre-wrap;
}

.emotion-0 b {
  font-weight: 600;
}

<span
  className="emotion-0"
>
  Pipeline execution failed: Function "bad_task" in pipeline "bad_task":
function_call_error: [string "bad_task"]:1: failure
stack traceback:
	[string "bad_task"]:1: in main chunk
	[C]: in function 'xpcall'
	.../errors.lua:148: in function 'pcall'
	.../init.lua:109: in function 'call'
	.../pipeline/init.lua:21: in function 'call_function'
	...pipeline/init.lua:63: in function 'run'
	.../runner/server.lua:70: in function &lt;.../runner/server.lua:68&gt;
stack traceback:
	.../pipeline/init.lua:25: in function 'call_function'
	.../pipeline/init.lua:63: in function 'run'
	.../runner/server.lua:70: in function &lt;...s/runner/server.lua:68&gt;
</span>
`;
